dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java

```diff
@@ -271,6 +279,33 @@ public VersionTypeValidation validateEntrySet(Set<SourceFile> sourcefiles, Strin
         return new VersionTypeValidation(true, null);
     }
 
+    public void checkForRecursiveHTTPImports(String content, Set<String> currentFileImports) throws IOException {
+        // Use matcher to get imports
+        String[] lines = StringUtils.split(content, '\n');
+
+        for (String line : lines) {
+            Matcher m = IMPORT_PATTERN.matcher(line);
+
+            while (m.find()) {
+                String match = m.group(1);
+                if (match.startsWith("http://") || match.startsWith("https://")) { // Don't resolve URLs
+                    System.out.println("Found match: " + match);
+                    if (currentFileImports.contains(match)) {
+                        throw new CustomWebApplicationException("Error parsing workflow. You may have a recursive import.",
+                                HttpStatus.SC_BAD_REQUEST);
+                    } else {
+                        System.out.println("New match: " + match);
+                        currentFileImports.add(match);
+                        URL url = new URL(match);
+                        String s = IOUtils.toString(url, StandardCharsets.UTF_8);
```

Comment:
Just to be paranoid and could be tracked separately from this PR, but if malicious user pointed to a multi-GB, this might bring down the web service with out of memory errors.